### ISSUE TITLE
MT5ForConditionalGeneration has model.config.max_length=20 by default. Why?

### ISSUE THREAD

**Issue by JoaoLages** [huggingface/transformers#16664]
- `transformers` version: 4.6.1
- Platform: Ubuntu 18
- Python version: 3.6

I spent one week training a T5 model with this package and couldn't figure out why my sequences obtained with Trainer were only yielding maximum of 20 tokens. I sent the `max_length` argument to the tokenizer to encode input/output method.
After a long time I found out that this happens:
```
model = MT5ForConditionalGeneration.from_pretrained('google/mt5-small')
model.config.max_length
Out: 20
```
The generate method was being used in Trainer because I used `predict_with_generate=True`. 
**Please change this behaviour, this was a very hard bug to find.  `model.config.max_length` should be set to `None` by default, if the model does not have limitations.**

**Comment by JoaoLages** [comment]
> Would it help if the console showed why generation stopped?
Would still be hard to find why `max_length` was reached. 

I still think it's best to always have `max_length` defined by default to the maximum value that the model can handle. A warning could be shown to know that this behaviour is in usage and that it may consume a lot of resources.

**Comment by JoaoLages** [comment]
I have the idea that no one is using early stopping on these models because 'it doesn't work' and it is due to this silent behaviour 😅

**Comment by JoaoLages** [comment]
Not really, `generate` works perfectly, it was just this silent small max_length setting :)
Some warnings/log messages would be better than nothing!

**Comment by gante** [comment]
(going to keep the issue open to backlink in a future improved logging PR)

**Comment by patrickvonplaten** [comment]
People that are familiar with `generate()` should know that `max_length` can and should be overwritten. I'll try to make the docs better here, but I don't think we should add a warning as this will literally be shown everytime someone calls generate without defining `max_length`
